core/Pos/Util/Util.hs

```diff
@@ -189,6 +191,18 @@ instance ToJSON Byte where
 instance IsString s => MonadFail (Either s) where
     fail = Left . fromString
 
+instance Rand.MonadRandom m => Rand.MonadRandom (ReaderT r m) where
```

Comment:
Nice. Though i think there was a similar instance in `test/Test/Pos/Block/...` somewhere. Can you check it?